tests/notes/notes.c

```diff
@@ -73,6 +73,144 @@ static int note_list_cb(
+		}
+	}
+
+	cl_assert_equal_i(1, retrieved_notes);
+}
+
+static struct note_create_payload can_create_a_note_from_commit[] = {
+	{ "1c9b1bc36730582a42d56eeee0dc58673d7ae869", "4a202b346bb0fb0db7eff3cffeb3c70babbd2045", 0 },
+	{ NULL, NULL, 0 }
+};
+
+void test_notes_notes__can_create_a_note_from_commit(void)
+{
+	unsigned int retrieved_notes = 0;
+	git_oid oid;
+	git_oid notes_commit_out;
+	git_reference *ref;
+	size_t i;
+
+	cl_git_pass(git_oid_fromstr(&oid, can_create_a_note_from_commit[0].object_oid));
+
+	cl_git_pass(git_note_commit_create(&notes_commit_out, NULL, _repo, NULL, _sig, _sig, &oid, "I decorate 4a20\n", 1));
+
+	/* create_from_commit will not update any ref,
+	 * so we must manually create the ref, that points to the commit */
+	cl_git_pass(git_reference_create(&ref, _repo, "refs/notes/i-can-see-dead-notes", &notes_commit_out, 0, NULL));
+
+	cl_git_pass(git_note_foreach(_repo, "refs/notes/i-can-see-dead-notes", note_list_create_cb, &can_create_a_note_from_commit));
+
+	for (i = 0; can_create_a_note_from_commit[i].note_oid != NULL; i++) {
+		if (can_create_a_note_from_commit[i].seen) {
+			retrieved_notes++;
+		}
+	}
+
+	cl_assert_equal_i(1, retrieved_notes);
+
+	git_reference_free(ref);
+}
+
+static struct note_create_payload can_create_a_note_from_commit_given_an_existing_commit[] = {
+	{ "1c9b1bc36730582a42d56eeee0dc58673d7ae869", "4a202b346bb0fb0db7eff3cffeb3c70babbd2045", 0 },
+	{ "1aaf94147c21f981e0a20bf57b89137c5a6aae52", "9fd738e8f7967c078dceed8190330fc8648ee56a", 0 },
+	{ NULL, NULL, 0 }
+};
+
+/* Test that we can create a note from a commit, given an existing commit */
+void test_notes_notes__can_create_a_note_from_commit_given_an_existing_commit(void)
+{
+	unsigned int retrieved_notes = 0;
+	git_oid oid;
+	git_oid notes_commit_out;
+	git_commit *existing_notes_commit = NULL;
+	git_reference *ref;
+	size_t i;
+
+	cl_git_pass(git_oid_fromstr(&oid, "4a202b346bb0fb0db7eff3cffeb3c70babbd2045"));
+
+	cl_git_pass(git_note_commit_create(&notes_commit_out, NULL, _repo, NULL, _sig, _sig, &oid, "I decorate 4a20\n", 0));
+
+	cl_git_pass(git_oid_fromstr(&oid, "9fd738e8f7967c078dceed8190330fc8648ee56a"));
+
+	git_commit_lookup(&existing_notes_commit, _repo, &notes_commit_out);
+
+	cl_assert(existing_notes_commit);
+
+	cl_git_pass(git_note_commit_create(&notes_commit_out, NULL, _repo, existing_notes_commit, _sig, _sig, &oid, "I decorate 9fd7\n", 0));
+
+	/* create_from_commit will not update any ref,
+	 * so we must manually create the ref, that points to the commit */
+	cl_git_pass(git_reference_create(&ref, _repo, "refs/notes/i-can-see-dead-notes", &notes_commit_out, 0, NULL));
+
+	cl_git_pass(git_note_foreach(_repo, "refs/notes/i-can-see-dead-notes", note_list_create_cb, &can_create_a_note_from_commit_given_an_existing_commit));
+
+	for (i = 0; can_create_a_note_from_commit_given_an_existing_commit[i].note_oid != NULL; i++) {
+		if (can_create_a_note_from_commit_given_an_existing_commit[i].seen) {
+			retrieved_notes++;
+		}
+	}
```

Comment:
Maybe a common function `assert_notes_seens` would be cool here, which simply asserts that all `.seen` fields are set.